public/src/modules/uploader.js

```diff
@@ -107,9 +116,11 @@ define('uploader', ['csrf', 'translator'], function(csrf, translator) {
 		return response;
 	}
 
-	module.hideAlerts = function(modal) {
-		$(modal).find('#alert-status, #alert-success, #alert-error, #upload-progress-box').addClass('hide');
-	};
+	function hasValidFileSize(fileElement, maxSize) {
+		if (window.FileReader && maxSize) {
+			return fileElement.files[0].size <= maxSize * 1000;
+		}
```

Comment:
[L69](https://github.com/NodeBB/NodeBB/pull/4479/files#diff-8492f30a9acf5c0dd88bc3f8707e343fR69) compares the return value explicitly to `false`. My logic is that only a truely too big file returns false, thus triggering the showAlert('error',..). Cases of maxSize not given or FileReader not present return `undefined`.